wifi-subsys/components/tools/src/serialization.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2022 Mesh4all <mesh4all.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @brief       Cbor functions to encode and decode values
+ *
+ * @author      Locha Mesh Developers (contact@locha.io)
+ *
+ */
+#include  <stdio.h>
+#include "serialization.h"
+#include "cbor.h"
+#include "esp_log.h"
+
+static const char* TAG = "SERIALIZATION";
+CborValue it;
+uint8_t buf[100];
+
+esp_err_t encode(uint8_t* coded_msg)
```

Comment:
There is only one input pointer, we need an output pointer and the size of the output pointer.